File: python_leetcode/ValidAnagram.py
class Solution(object):
    def isAnagram(self, s: str, t: str) -> bool:
        """Checks if two strings are Anagrams.
        Two strings are considered Anagrams if they are made from the same no. of each letter
        Youtube explanation : https://www.youtube.com/watch?v=9UtInBqnCgA&t=640s

        Args:
            s (str): string 1
            t (str): string 2

        Returns:
            bool: True if Anagram, else False
        """

        if len(s) != len(t):
            return False

        countS, countT = {}, {}

        for i in range(len(s)):
            countS[s[i]] = 1 + countS.get(s[i], 0)
            countT[t[i]] = 1 + countT.get(t[i], 0)
        return countS == countT

File: cpp_leetcode/valid_anagram.cpp
#include <iostream>
#include <string>
#include <unordered_map>

using namespace std;

class Solution
{
public:
    /**
     * @brief Leetcode 242: Check if two strings are anagrams of each other.
     * Two strings are anagrams if they contain the same characters but in a different order.
     */
    bool isAnagram(string s, string t)
    {
        if (s.size() != t.size())
        {
            return false;
        }

        unordered_map<char, int> count_s;
        unordered_map<char, int> count_t;

        for (int i = 0; i < s.size(); i++)
        {
            count_s[s[i]]++;
            count_t[t[i]]++;
        }

        return count_s == count_t;
    }
};

int main()
{
    Solution solution;
    string s = "anagram";
    string t = "nagaram";

    if (solution.isAnagram(s, t))
    {
        cout << "Is valid anagram.";
    }
    else
    {
        cout << "Is not valid anagram.";
    }

    return 0;
}